grpc/src/main/java/io/stargate/grpc/service/CreateTableHandler.java

```diff
@@ -0,0 +1,245 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletionStage;
+
+class CreateTableHandler extends MessageHandler<CqlTableCreate> {
+
+  private final Persistence.Connection connection;
+  private final String query;
+
+  CreateTableHandler(
+      CqlTableCreate createTable,
+      Persistence.Connection connection,
+      Persistence persistence,
+      QueryBuilder queryBuilder,
+      StreamObserver<QueryOuterClass.Response> responseObserver) {
+    super(createTable, responseObserver);
+    this.connection = connection;
+    this.query = buildQuery(createTable, persistence, queryBuilder);
+  }
+
+  @Override
+  protected void validate() {
+    // intentionally empty
+  }
+
+  @Override
+  protected CompletionStage<QueryOuterClass.Response> executeQuery() {
+    long queryStartNanoTime = System.nanoTime();
+    try {
+      return connection
+          .execute(
+              new SimpleStatement(query),
+              makeParameters(connection.clientInfo()),
+              queryStartNanoTime)
+          .thenApply(result -> makeResponseBuilder(result).build());
+    } catch (Exception e) {
+      return failedFuture(e, false);
+    }
+  }
+
+  private String buildQuery(
+      CqlTableCreate createProto, Persistence persistence, QueryBuilder queryBuilder) {
+    String keyspaceName =
+        persistence.decorateKeyspaceName(
+            createProto.getKeyspaceName(), GrpcService.HEADERS_KEY.get());
+    CqlTable tableProto = createProto.getTable();
+
+    List<Column> columns = new ArrayList<>();
+    for (ColumnSpec columnProto : tableProto.getPartitionKeyColumnsList()) {
+      columns.add(
+          ImmutableColumn.create(
+              columnProto.getName(), Column.Kind.PartitionKey, convertType(columnProto.getType())));
+    }
+    for (ColumnSpec columnProto : tableProto.getClusteringKeyColumnsList()) {
+      String name = columnProto.getName();
+      Schema.ColumnOrderBy orderProto =
+          tableProto.getClusteringOrdersMap().getOrDefault(name, Schema.ColumnOrderBy.ASC);
+      columns.add(
+          ImmutableColumn.create(
+              name,
+              Column.Kind.Clustering,
+              convertType(columnProto.getType()),
+              convertOrder(orderProto)));
+    }
+    for (ColumnSpec columnProto : tableProto.getStaticColumnsList()) {
+      columns.add(
+          ImmutableColumn.create(
+              columnProto.getName(), Column.Kind.Static, convertType(columnProto.getType())));
+    }
+    for (ColumnSpec columnProto : tableProto.getColumnsList()) {
+      columns.add(
+          ImmutableColumn.create(
+              columnProto.getName(), Column.Kind.Regular, convertType(columnProto.getType())));
+    }
+
+    return queryBuilder
+        .create()
+        .table(keyspaceName, tableProto.getName())
+        .ifNotExists(createProto.getIfNotExists())
+        .column(columns)
+        // TODO handle table options
```

Comment:
If we want to support generic `string => string` options, we'll need to add that capability to `QueryBuilder`.

But if we aim for a similar API as v1, the only option that's exposed so far is the default TTL, so we could go with that and just have a `defaultTimeToLIve` parameter in the gRPC message.

Reply:
I think we should treat table options (and keyspace options) as map<string, string> on the gRPC API so that the values can be obtained for the sake of completeness, in case there is a need to allow API consumers to obtain the full definitions of keyspaces and tables. 

However, I doubt there are many cases where the API clients will need to process the actual values, so I'm hesitant to put the effort into modeling them in the gRPC API. This seemed like a good compromise approach to me.